package/models.py

```diff
@@ -470,6 +471,23 @@ def last_commit(self):
     def commits_over_52_listed(self):
         return [int(x) for x in self.commits_over_52().split(",")]
 
+class FlaggedPackage(BaseModel):
+    package = models.ForeignKey(Package, on_delete=models.CASCADE, related_name="flags")
+    reason = models.TextField()
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    created = models.DateTimeField(auto_now_add=True)
+    approved_flag = models.BooleanField(default=False)
+
+    def approve(self):
+        self.approved_flag = True
+        self.save()
+
+    class Meta:
+        ordering = ["-created"]
+        UniqueConstraint(fields=["package", "user"], name="unique_flagged_package")
```

Comment:
I think this would need to be activated with `constraints = [UniqueConstraint(fields=["package", "user"], name="unique_flagged_package")]`.